plugins/node/instrumentation-tedious/test/api.ts

```diff
@@ -16,17 +16,31 @@
 
 import * as assert from 'assert';
 import { promisify } from 'util';
-import type { Connection, Request, TYPES, ConnectionConfig } from 'tedious';
+import * as semver from 'semver';
+
+import type {
+  Connection,
+  Request,
+  TYPES,
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  ConnectionConfig as ConnectionConfigLegacy,
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  ConnectionConfiguration as ConnectionConfigNew,
```

Comment:
I know this is not ideal. I was not able to find a better & easier solution. And I am not very familiar with TS.

Possible long-term:
- write tests in TS
- run tests in JS
- requirement: remove the test build folder from the git ignore list
- needs a follow up ticket IMO